src/network.rs

```diff
@@ -116,16 +199,343 @@ impl Nectar {
+                            "could order_id corresponding to shared_swap_id: {}",
+                            shared_swap_id
+                        );
+                        return;
+                    }
+                };
+
+                // TODO: Handle expiries properly when orderbook does so
+
+                let hbit_params = hbit::Params::new(
+                    hbit::SharedParams {
+                        asset: asset::Bitcoin::from_sat(satoshi_amount),
+                        redeem_identity: maker_bitcoin_identity,
+                        refund_identity: taker_bitcoin_identity,
+                        expiry: Timestamp::from(absolute_expiry),
+                        secret_hash,
+                        // TODO: Make it dynamic when orderbook handles networks
+                        network: ::bitcoin::Network::Regtest,
+                    },
+                    bitcoin_transient_sk,
+                );
+
+                let herc20_params = herc20::Params {
+                    asset: erc20_asset,
+                    redeem_identity: taker_ethereum_identity,
+                    refund_identity: maker_ethereum_identity,
+                    expiry: Timestamp::from(absolute_expiry),
+                    secret_hash,
+                    // TODO: Make it dynamic once orderbook handles networks
+                    chain_id: ChainId::regtest(),
+                };
+
+                let swap = SwapKind::HbitHerc20(SwapParams {
+                    hbit_params,
+                    herc20_params,
+                    secret_hash,
+                    start_of_swap: Utc::now().naive_local(),
+                    swap_id: SwapId::default(),
+                });
+
+                self.events.push_back(Event::SpawnSwap(swap, taker));
+            }
+        }
+    }
+}
+
+struct TokioExecutor {
+    handle: tokio::runtime::Handle,
+}
+
+impl libp2p::core::Executor for TokioExecutor {
+    fn exec(&self, future: Pin<Box<dyn Future<Output = ()> + Send>>) {
+        let _ = self.handle.spawn(future);
+    }
+}
+
+#[derive(Debug)]
+pub struct TakenOrder {
+    pub id: OrderId,
+    pub inner: BtcDaiOrder,
+    pub taker: Taker,
+    confirmation_channel: ResponseChannel<take_order::Response>,
+}
+
+impl TakenOrder {
+    fn new(
+        order: comit::network::orderbook::Order,
+        taker_id: PeerId,
+        confirmation_channel: ResponseChannel<take_order::Response>,
+    ) -> Self {
+        // TODO: comit::network::orderbook does not yet support selling Bitcoin
+        let inner = BtcDaiOrder {
+            position: Position::Buy,
+            base: bitcoin::Amount::from_sat(order.buy),
+            quote: order.sell.into(),
+        };
+
+        let taker = Taker::new(taker_id);
+
+        Self {
+            id: order.id,
+            inner,
+            taker,
+            confirmation_channel,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct Taker(PeerId);
```

Comment:
Lol, we both did the same change but slightly different... :/

Reply:
Stuck with mine after merging because it was easier, but we can change it back to yours.

Reply:
Mine was just to make it easy for you to add the fields that you need :)

Reply:
Reverted it back to your design :D